kite-data/kite-data-core/src/main/java/org/kitesdk/data/View.java

```diff
@@ -128,4 +128,41 @@
    * @since 0.16.0
    */
   public URI getUri();
+
+  /**
+   * Signal that the view is ready for consumption.
+   *
+   * May result in a no-op if the view implementation does not support the ready
+   * concept.
+   *
+   * Note that an {@link #isEmpty() empty} view may be signaled as ready.
+   *
+   * @throws UnsupportedOperationException if the view does not support the ready
+   *          concept.
```

Comment:
I like the idea of using an interface to carry these methods. If we decide later that all views should be `Signalable` (or whatever it turns out to be) then it's a binary compatible change to have `View` extend that interface. Removing those methods is not a compatible change.

Another reason for the interface is that we generally prefer compile-time problems to runtime problems. If the method always exists, then it is less likely that a caller is going to handle the `UnsupportedOperationException` case. But if you have to check for an interface and then cast, the caller has to account for views that are not signalable.